piton/design/chip/tile/rtl/tile.v.pyv

```diff
@@ -827,289 +827,57 @@ if (TILE_TYPE == `BLACKPARROT_TILE) begin : g_blackparrot_core
-        .dsync_o                            (dsync_lo),
-        .ready_o                            (transducer_ready_lo),
-        // OpenPiton Bus
-        // BP -> L1.5
-        .miss_v_i                           (arb_miss_v_lo),
-        .miss_yumi_o                        (arb_miss_yumi_li),
-        .uncached_i                         (arb_uncached_lo),
-        .miss_addr_i                        (arb_miss_addr_lo),
-        .lru_way_i                          (arb_miss_lru_lo),
-        .store_i                            (arb_is_store_lo),
-        .store_data_i                       (arb_miss_data_lo),
-        .size_op_i                          (arb_miss_size_lo),
-
-        // L1.5 -> BP
-        .l15_transducer_ack                 (l15_transducer_ack),
-        .l15_transducer_header_ack          (l15_transducer_header_ack),
-
-        .transducer_l15_rqtype              (transducer_l15_rqtype),
-        .transducer_l15_amo_op              (transducer_l15_amo_op),
-        .transducer_l15_size                (transducer_l15_size),
-        .transducer_l15_val                 (transducer_l15_val),
-        .transducer_l15_address             (transducer_l15_address),
-        .transducer_l15_data                (transducer_l15_data),
-
-        .transducer_l15_nc                  (transducer_l15_nc),
-        .transducer_l15_threadid            (transducer_l15_threadid),
-        .transducer_l15_prefetch            (transducer_l15_prefetch),
-        .transducer_l15_blockstore          (transducer_l15_blockstore),
-        .transducer_l15_blockinitstore      (transducer_l15_blockinitstore),
-        .transducer_l15_l1rplway            (transducer_l15_l1rplway),
-        .transducer_l15_invalidate_cacheline(transducer_l15_invalidate_cacheline),
-        .transducer_l15_csm_data            (transducer_l15_csm_data),
-        .transducer_l15_data_next_entry     (transducer_l15_data_next_entry),
-
-        .l15_transducer_val                 (l15_transducer_val),
-        .l15_transducer_returntype          (l15_transducer_returntype),
-
-        .l15_transducer_data_0              (l15_transducer_data_0),
-        .l15_transducer_data_1              (l15_transducer_data_1),
-
-        .transducer_l15_req_ack             (transducer_l15_req_ack),
-
-        .data_mem_pkt_o                     (arb_data_mem_pkt_li),
-        .data_mem_pkt_v_o                   (arb_data_mem_pkt_v_li),
-        .data_mem_pkt_yumi_i                (arb_data_mem_pkt_yumi_lo),
-
-        .tag_mem_pkt_o                      (arb_tag_mem_pkt_li),
-        .tag_mem_pkt_v_o                    (arb_tag_mem_pkt_v_li),
-        .tag_mem_pkt_yumi_i                 (arb_tag_mem_pkt_yumi_lo),
-
-        .stat_mem_pkt_o                     (arb_stat_mem_pkt_li),
-        .stat_mem_pkt_v_o                   (arb_stat_mem_pkt_v_li),
-        .stat_mem_pkt_yumi_i                (arb_stat_mem_pkt_yumi_lo)
-    );
+    assign spc_grst_l = reset_rrrr;
```

Comment:
what's this do?

Reply:
This is the reset that goes to the L1.5. It was there as a pico reset module in your implementation. We ran into some issues because the tool wasn't finding the pico_reset module so we just put the logic in that module into this place.

Reply:
There's a flopped reset (spc_grst_l) that gets passed from the core into the L1.5. For pico this comes in a pico_reset module. We could instantiate that here if we move pico_reset into piton/design/common/ (I've done this in our internal repo) but the solution here is fine too

Reply:
Maybe a comment // reset the L1.5.  This must be flopped N cycles after initial reset

Also, what are the timing requirements here?

Reply:
I actually don't have an answer on that. The core shouldn't drive valid on TRI until it gets the reset packet anyway so I'm not entirely sure it matters. I think the OST1 just wanted its CCX to be reset according to a reset that it provided. Honestly that doesn't make much sense to me though because it had eight cores. Maybe it's more for its FPU (also connected by CCX), I'm not sure.